apps/snitch_api/lib/snitch_api_web/controllers/product_controller.ex

```diff
@@ -71,17 +48,17 @@ defmodule SnitchApiWeb.ProductController do
   end
 
   def show(conn, %{"product_slug" => slug} = params) do
-    case Context.product_by_slug(slug) do
```

Comment:
@anantanant2015 this change is not required.

Reply:
`show` method should be just used to fetch a single product. We should not confuse this with the filtering api and not try to make it common with context methods. Please make these changes.

Reply:
done